system/lib/pthread/library_pthread.c

```diff
@@ -913,6 +913,11 @@ int llvm_atomic_load_add_i32_p0i32(int* ptr, int delta) {
   return emscripten_atomic_add_u32(ptr, delta);
 }
 
+// Stores the memory address that the main thread is waiting on, if any. If
+// the main thread is waiting, we wake it up before waking up any workers.
+EMSCRIPTEN_KEEPALIVE
+int main_thread_futex;
```

Comment:
Why `int` if this stores a memory address? Or is it the address of this data that is useful?

Reply:
Oh you're right, this should be a `void*`, fixed.

While in general futexes are always 32 bits, this is not an actual futex, it's the address of the futex (which some day soon may be 64 bits).